core/domain/user_jobs_one_off_test.py

```diff
@@ -1585,6 +1585,272 @@ def test_one_setting_model_without_gae_id(self):
+        )
+        user_settings_model.update_timestamps()
+        original_created_on_timestamp = user_settings_model.created_on
+        original_last_updated_timestamp = user_settings_model.last_updated
+        user_settings_model.created_on += datetime.timedelta(hours=10)
+        user_settings_model.last_updated += datetime.timedelta(hours=10)
+        user_settings_model.put()
+
+        user_auth_details_model = (
+            user_models.UserAuthDetailsModel(
+                id=self.USER_ID_1,
+                gae_id='gae_id'
+            )
+        )
+        user_auth_details_model.update_timestamps()
+        user_auth_details_model.put()
+
+        learner_playlist_model = user_models.LearnerPlaylistModel(
+            id=self.USER_ID_1,
+            exploration_ids=self.EXPLORATION_IDS,
+            collection_ids=self.COLLECTION_IDS,
+        )
+        learner_playlist_model.update_timestamps()
+        learner_playlist_model.put()
+
+        user_skill_mastery_model = user_models.UserSkillMasteryModel(
+            id=user_models.UserSkillMasteryModel.construct_model_id(
+                self.USER_ID_1, self.SKILL_ID_1),
+            user_id=self.USER_ID_1,
+            skill_id=self.SKILL_ID_1,
+            degree_of_mastery=self.DEGREE_OF_MASTERY
+        )
+        user_skill_mastery_model.update_timestamps()
+        user_skill_mastery_model.put()
+
+        final_created_on_timestamp = user_auth_details_model.created_on
+
+        self.assertNotEqual(
+            user_settings_model.created_on, original_created_on_timestamp)
+        self.assertNotEqual(
+            user_settings_model.last_updated, original_last_updated_timestamp)
+        self.assertNotEqual(
+            user_settings_model.created_on, final_created_on_timestamp)
+
+        output = self._run_one_off_job()
+        # There already exists a UserSettings created in test_utils. It's
+        # corresponding user_auth_details_model is created first (which has
+        # a lower timestamp), hence it is also being updated here.
+        self.assertItemsEqual([['SUCCESS_UPDATED', 2]], output)
+
+        migrated_user_model = (
+            user_models.UserSettingsModel.get_by_id(self.USER_ID_1))
+
+        self.assertEqual(
+            migrated_user_model.created_on, final_created_on_timestamp)
+
+    def test_updated_using_other_datetime_attributes_of_other_models(self):
+        user_settings_model = (
+            user_models.UserSettingsModel(
+                id=self.USER_ID_1,
+                email=self.EMAIL_1,
+            )
+        )
+        user_settings_model.update_timestamps()
+        original_created_on_timestamp = user_settings_model.created_on
+        user_settings_model.created_on += datetime.timedelta(hours=10)
+        user_settings_model.last_updated += datetime.timedelta(hours=10)
+        user_settings_model.put()
+
+        final_created_on_timestamp = (
+            original_created_on_timestamp + datetime.timedelta(hours=1))
```

Comment:
I think doing `final_created_on_timestamp = user_subscriptions_model.last_checked` down below and instead of `last_checked=final_created_on_timestamp` do `last_checked=original_created_on_timestamp + datetime.timedelta(hours=1)`

Reply:
It still functions in the same way. I did it my way so that I define what the `final_created_on_timestamp` would be in the starting, and then assign all the other date-time parameters according to it. If you are suggesting this to improve the readability, I will do the change. But I would like to know if there's any other reason apart from that. Thanks!